### PR TITLE
MAINT: In the cat subcommand, replace the usage of the deprecated PdfMerger by PdfWriter

### LINKED ISSUE #31: In the cat subcommand, replace the usage of the deprecated PdfMerger class from pypdf
`PdfMerger` is used in the in the `cat` subcommand: https://github.com/py-pdf/pdfly/blob/main/pdfly/cat.py#L51

But this class is deprecated in `pypdf` since version 3.9.1: https://github.com/py-pdf/pypdf/releases/tag/3.9.1

We should replace it by combining `PdfReader`, its `merge_page()` method & `PdfWriter` from `pypdf`.
An example of code using those can be found there: https://py-pdf.github.io/fpdf2/CombineWithPypdf.html

Extra unit tests could be added in the process in https://github.com/py-pdf/pdfly/blob/main/tests/test_cat.py
in order to ensure that `pdfly cat` still works similarly as it currently does.

### PR DESCRIPTION
**Pull request by kommade**
Removed deprecated PdfMerger as in #31 and replaced with PdfReader and PdfWriter. No changes made to [test_case](https://github.com/py-pdf/pdfly/blob/main/tests/test_cat.py) as functionality seems to be exactly the same.

Closes #31

### CODE REVIEW

**Review by Lucas-C — APPROVED** [review]

**Comment by Lucas-C** [comment]
Good job @kommade!

I just approved this PR.
I'll let @MartinThoma review those changes as well before merging,
as he may want to perform extra checks himself to ensure the behaviour of `pdfly` has not changed,
or maybe suggest new unit tests.

By the way, are you contributing as part of https://hacktoberfest.com @kommade?

**Comment by Lucas-C** [comment]
@MartinThoma : do you think that this could be merged? 🙂

File: pdfly/cat.py
Comment (MartinThoma) [inline pdfly/cat.py:77]:
```suggestion
                writer.add_page(reader.pages[page_num])
```

**Comment by MartinThoma** [comment]
@kommade I'm very sorry that it took me so long to review / merge this PR :see_no_evil: 

I remember that I had a quick glance, but wanted to add a few tests. Then I didn't have time for that and forgot :sweat_smile: 
Always feel free to gently remind me of PRs - just like Lucas did :-) 

Good work with the PR! There was a small off-by-one that was easy to fix with more tests.

Your contribution will be on PyPI latest tomorrow :-)

**Comment by MartinThoma** [comment]
@Lucas-C Thank you for pinging me about this :hugs:  And also for your help with the other issues/PRs/discussions :heart:

**Comment by kommade** [comment]
Hi! I actually just wanted to try contributing to public projects on github and it just happened to be hacktoberfest too! I'm glad it was accepted, no worries on the delay I wasn't too concerned anyway

**Pull request merged**
